integration/class-litespeed-cache.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * LiteSpeed Cache integration file.
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub\Integration;
+
+/**
+ * LiteSpeed Cache integration.
+ *
+ * @see https://wordpress.org/support/topic/avoiding-caching-activitypub-content/
+ */
+class Litespeed_Cache {
+
+	/**
+	 * The rules to add to the htaccess file.
+	 *
+	 * @var string
+	 */
+	private static $rules = '<IfModule LiteSpeed>
+RewriteEngine On
+RewriteCond %{HTTP:Accept} application
+RewriteRule ^ - [E=Cache-Control:vary=%{ENV:LSCACHE_VARY_VALUE}+isjson]
+</IfModule>';
+
+	/**
+	 * The option name to store the htaccess rules.
+	 *
+	 * @var string
+	 */
+	private static $option_name = 'activitypub_integration_litespeed_cache_htaccess_rules';
+
+	/**
+	 * The marker to identify the rules in the htaccess file.
+	 *
+	 * @var string
+	 */
+	private static $marker = 'ActivityPub LiteSpeed Cache';
+
+	/**
+	 * Initialize the integration.
+	 */
+	public static function init() {
+		\add_action( 'activate_litespeed-cache/litespeed-cache.php', array( self::class, 'add_htaccess_rules' ) );
+		\add_action( 'deactivate_litespeed-cache/litespeed-cache.php', array( self::class, 'remove_htaccess_rules' ) );
+
+		\add_filter( 'site_status_tests', array( self::class, 'maybe_add_site_health' ) );
+	}
+
+	/**
+	 * Add the Litespeed Cache htaccess rules.
+	 */
+	public static function add_htaccess_rules() {
+		// Ensure get_home_path() is declared.
+		require_once ABSPATH . 'wp-admin/includes/file.php';
+
+		$htaccess_file = get_home_path() . '.htaccess';
+		$added_rules   = false;
+
+		if ( \wp_is_writable( $htaccess_file ) ) {
+			$added_rules = \insert_with_markers( $htaccess_file, self::$marker, self::$rules );
+		}
+
+		\update_option( self::$option_name, $added_rules );
+	}
+
+	/**
+	 * Remove the Litespeed Cache htaccess rules.
+	 */
+	public static function remove_htaccess_rules() {
+		// Ensure get_home_path() is declared.
+		require_once ABSPATH . 'wp-admin/includes/file.php';
+
+		$htaccess_file = get_home_path() . '.htaccess';
+
+		if ( \wp_is_writable( $htaccess_file ) ) {
+			\insert_with_markers( $htaccess_file, self::$marker, '' );
+		}
+
+		\delete_option( self::$option_name );
+	}
+
+	/**
+	 * Maybe add the Litespeed Cache config to the site health.
+	 *
+	 * @param array $tests The site health tests.
+	 *
+	 * @return array The site health tests with the Litespeed Cache config test.
+	 */
+	public static function maybe_add_site_health( $tests ) {
+		if ( ! \is_plugin_active( 'litespeed-cache/litespeed-cache.php' ) ) {
+			return $tests;
+		}
+
+		$tests['direct']['activitypub_test_litespeed_cache_integration'] = array(
```

Comment:
It would be good to test whether we're on Apache before adding the test.

Reply:
this is how core does it so I thought I go with the flow ;) !?!